src/viewer.js

```diff
@@ -124,14 +139,47 @@ class Viewer {
   }
   
   /**
-   * @param {?string} urlPath
+   * @param {boolean} isLastBack true if back button was hit and viewer should hide.
+   * @param {number} opt_poppedAmpHistoryIndex The AMP History Index that was just popped.
    * @private
     */
-  handleChangeHistoryState_(urlPath) {
-    if (urlPath) {
-      if (this.showViewer_) this.showViewer_();
-    } else {
-      if (this.hideViewer_) this.hideViewer_();
+  handleChangeHistoryState_(isLastBack, opt_poppedAmpHistoryIndex) {
+    if (this.showViewer_ && this.isViewerHidden_ && this.isViewerHidden_()) {
+      this.showViewer_();
+      return;
+    }
+    if (isLastBack && this.hideViewer_) {
```

Comment:
I think you want to return even if there is no `this.hideViewer_`. A history pop that closes the viewer should never to sent to the AMP doc.